### DISCUSSION TITLE
How to select different processes based on the number of files in channel.fromPath?

### DISCUSSION THREAD

**Discussion by ChenDepp** [nextflow-io/nextflow#5258] [Q&A]
hi:
    How to select different processes based on the number of files in channel.fromPath?
   Can you tell me how to achieve it? 
   hava a good day!

**Comment by mribeirodantas** [comment]
Create a folder, and store the following Nextflow code to a file named `script.nf`:

```
process A {
  debug true

  input:
  path filepath

  output:
  stdout

  script:
  """
  echo Less than 3: ${filepath}
  """
}

process B {
  debug true

  input:
  path filepath

  output:
  stdout

  script:
  """
  echo 3 or more: ${filepath}
  """
}

workflow {
  Channel
    .fromPath("*.txt")
    .set { my_ch }

  my_ch
    .collect()
    .branch {
        small: it.size() < 3
        large: it.size() >= 3
    }
    .set { result }

  A(result.small.flatMap())
  B(result.large.flatMap())
}
```

Create 2 .txt files (e.g. `touch a.txt; touch b.txt`) and run `nextflow run script.nf` and see what happens. Then create a third file (e.g. `touch c.txt`) and run it again. As you can see, we're using a different process depending on the number of elements in the channel created with the `fromPath` channel factory. Depending on what you want to do, you may need to adapt this code.

Also, please next time ask at https://community.seqera.io
